contributor-docs/deprecating-components.md

```diff
@@ -0,0 +1,77 @@
+  - [ ] Make sure to update the import code block. I.e. `import { ActionList } from "@primer/react/deprecated"`.
+- [ ] Add a `Deprecation` section with the link of the newly developed component and provide a diff. See [deprecated ActionList docs](https://primer.style/react/deprecated/ActionList#deprecation) as an example.
+- [ ] Move the new components docs from draft folder to the main docs folder `docs/content/` and update the title by removing `v2`, status as well as the storybook and the source code link.
+  - [ ] Make sure to update `jsx live drafts` -> `jsx live`
+  - [ ] Make sure to update the import code block. I.e. `import { ActionList } from "@primer/react"`.
+  - [ ] Update the navigation on `docs/src/@primer/gatsby-theme-doctocat/nav.yml` accordingly.
+
+### Comms side of the deprecation
+
+When we know if the newly developed component is ready to go in the main bundle and the old component is good to be deprecates in the next major release, we announce the deprecation and the promotion in the [Primer changelog](https://github.com/github/primer/discussions/categories/primer-changelog) (GitHub staff only) under the `Coming soon` section.
+
+To ensure the impending deprecation is included in the Primer changelog, Primer maintainers should highlight any upcoming deprecations in their [weekly status updates](https://github.com/github/design-infrastructure/blob/main/how-we-work/planning-and-tracking-work/updates.md#weekly-status-updates-required) (GitHub staff only).
+
+Once the major release is out, we provide support on how to switch using the newly develop component as well as how to continue to use the deprecated component for folks who prefer to migrate later on.
+
+We also write codemods to deprecate the old component and use the new component under [our migration repo](https://github.com/primer/react-migrate#readme).
+
+### Primer maintainers to alert Accessibility team about deprecations
+
+As GitHub teams onboard to the [accessibility scorecard](https://github.com/github/engineering/discussions/2443) (GitHub staff only), we have a new opportunity to encourage teams to remove deprecated components and adopt new, more accessible ones.
+
+Primer maintainers should reach out to #accessibility any time a component is planned for deprecation, or if there is a new component that we want to enforce usage around.
+
+## Removing the deprecated component
+
+After about 6 months of living in the deprecated bundle, a component is retired/deleted from the codebase. This is also a breaking change and we release it in the upcoming major release.
+At this point, consumers are expected to plan migration work and we annouce the deletion in Primer Changelog as we do for the deprecation.
+
+## FAQ
+
+- Does the deprecated component accept new feature request?
+  - No, because we have a single bundle for all components, you can not pick the components you want to upgrade. This can result in additional unrelated work.
```

Comment:
Might be helpful to highlight that, once deprecated, the component is likely in a "maintenance mode" and will only receive bug fixes. Maybe something like:

```suggestion
  - No, once a component is deprecated the only changes it will receive are bug fixes.
```

Reply:
What do y'all think: should this be _will receive_ or _may receive_? Not sure how dedicated we've been to handling all bugfixes in deprecated components, or just major functional bugs 🤔

Reply:
I think "may receive" sounds good to me. I like that it removes the certainty.
 
I updated to be "may receive" but please let me know if anyone has a concern 🙌🏻